source/extensions/transport_sockets/tls/cert_validator/san_matcher_config.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/config/core/v3/extension.pb.h"
+#include "envoy/extensions/transport_sockets/tls/v3/common.pb.h"
+#include "envoy/ssl/certificate_validation_context_config.h"
+#include "envoy/type/matcher/v3/string.pb.h"
+
+#include "source/common/common/hash.h"
+#include "source/common/common/matchers.h"
+#include "source/common/protobuf/protobuf.h"
+#include "source/extensions/transport_sockets/tls/cert_validator/default_validator.h"
+#include "source/extensions/transport_sockets/tls/utility.h"
+
+#include "openssl/x509v3.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Tls {
+
+template <int general_name_type> class StringSanMatcher : public Envoy::Ssl::SanMatcher {
```

Comment:
It seems like overkill to template this; I think it would be simpler to just pass the `general_name_type` as a runtime parameter.

Reply:
Fixed.